src/core/shared_state/currentEventResource.ts

```diff
@@ -1,31 +1,22 @@
-import { createAtom } from '~utils/atoms';
+import { atom } from '@reatom/framework';
 import { apiClient } from '~core/apiClientInstance';
 import { createAsyncAtom } from '~utils/atoms/createAsyncAtom';
 import { i18n } from '~core/localization';
+import { v3toV2 } from '~utils/atoms/v3tov2';
 import { currentEventAtom } from './currentEvent';
 import { currentEventFeedAtom } from './currentEventFeed';
 import type { EventWithGeometry } from '~core/types';
 
-const eventDependencyAtom = createAtom(
-  { currentEventAtom },
-  (
-    { get, getUnlistedState },
-    state: { event: { id: string } | null; feed: { id: string } | null } = {
-      event: null,
-      feed: null,
-    },
-  ) => {
-    const event = get('currentEventAtom');
-    if (!event) return { event: null, feed: state.feed };
+const eventDependencyAtom = atom((ctx) => {
+  const event = ctx.get(currentEventAtom.v3atom);
```

Comment:
`ctx.get()` is not equivalent to v2 `get() `

V2 `get()` actually ubscribes to value changes, so you need `ctx.spy()` here

Reply:
Looks like this is what breaks updating Selected Area

Reply:
yes, thanks for investigation